### PR TITLE
ErrorTripSwitch implementation

### PR DESCRIPTION
**Pull request by kcrobinson**
@smarts @pikadanx 

Reimplmenetation of https://github.com/infospace/log4net.awsKinesis/pull/6 into develop-short. Includes refactor of singleton.

### CODE REVIEW

**Comment by smarts** [comment]
We talked about this offline, but it feels like you could make this an `IErrorHandler` implementation, which is nice since that's an interface within the log4net framework.

**Comment by kcrobinson** [comment]
Trip switch extracted into its own library: https://github.com/infospace/log4net.tripSwitchErrorHandler

Backend pull request to show how I hope this will all work: https://github.com/infospace/search_backend/pull/871

File: src/log4net.AwsKinesisAppender/DisabledAmazonKinesisClient.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+{
+    internal class DisabledAmazonKinesisClient : IAmazonKinesis
+    {
+        public void Dispose()
+        {
+        }
+
+        public AddTagsToStreamResponse AddTagsToStream(AddTagsToStreamRequest request)
+        {
+            throw new NotImplementedException();
```
Comment (smarts) [inline src/log4net.AwsKinesisAppender/DisabledAmazonKinesisClient.cs]: All the methods should return an instance instead of throwing an exception.

**Comment by smarts** [comment]
I made the new repo private since we never got permission for it to be public. Also, what do you think about renaming it to something less specific like `log4net.ErrorHandlers`?

**Comment by kcrobinson** [comment]
Closing: Moved to https://github.com/infospace/log4net.awsKinesis/pull/8

**Pull request closed without merging**